featuregate/gates.go

```diff
@@ -19,14 +19,54 @@ import (
 	"sync"
 )
 
+// Stage represents the Gate's lifecycle and what is the expected state of it.
+type Stage int8
+
+const (
+	// Alpha is used when creating a new feature and the Gate must be explicitly enabled
+	// by the operator.
+	//
+	// The Gate will be disabled by default.
+	Alpha Stage = iota
+	// Beta is used when the feature flag is well tested and is enabled by default,
+	// but can be disabled by a Gate.
+	//
+	// The Gate will be enabled by default.
+	Beta
+	// Stable is used when feature is permanently enabled and can not be disabled by a Gate.
+	// This value is used to provide feedback to the operator that the gate will be removed in the next version.
+	//
+	// The Gate will be enabled by default and will return an error if modified.
+	Stable
+)
+
+var stageNames = map[Stage]string{
+	Alpha:  "Alpha",
+	Beta:   "Beta",
+	Stable: "Stable",
+}
+
+func (s Stage) String() string {
+	if v, ok := stageNames[s]; ok {
+		return v
+	}
+	return "unknown"
+}
+
 // Gate represents an individual feature that may be enabled or disabled based
 // on the lifecycle state of the feature and CLI flags specified by the user.
 type Gate struct {
 	ID          string
 	Description string
-	Enabled     bool
+	// Deprecated: [v0.64.0] Use Stage instead, this field will move to be non exported.
+	Enabled        bool
+	Stage          Stage
+	IssueLinks     []string
```

Comment:
I think that makes sense, my thoughts were if a flag touches multiple issues then it would be nice to reference each of them but you could do the same on the flag issue as well.

I'm not fixed on the name, so I am happy to change it.